tests/NuGet.Services.AzureSearch.Tests/SearchService/SearchTextBuilderFacts.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace NuGet.Services.AzureSearch.SearchService
+{
+    public class SearchTextBuilderFacts
+    {
+        public class V2Search : FactsBase
+        {
+            [Theory]
+            [MemberData(nameof(CommonAzureSearchQueryData))]
+            public void GeneratesAzureSearchQuery(string input, string expected)
+            {
+                var actual = _target.V2Search(new V2SearchRequest { Query = input });
+
+                Assert.Equal(expected, actual);
+            }
+
+            // TODO: id and packageId query fields should map to different fields with different analyzers
+            // See: https://github.com/NuGet/NuGetGallery/issues/6920
+            // See: https://github.com/NuGet/NuGetGallery/issues/6922
+            [Theory]
+            [InlineData(false, "packageId:hello")]
```

Comment:
This should only happen if `luceneQuery` is `true` right?

Reply:
That's correct. I'm unit testing the current behavior, but this will be addressed by the TODO issues that are linked in the comment above.